### PR TITLE
Remove initiallyInRotation=false option: Not used

### PR DESCRIPTION
**Pull request by bratseth**
@gjoranv please review

I suggest reviewing each commit in order. The purpose of this is to tie the state/v1/health response of the container to the vip status: Whenever the container decides that it is not fit to receive traffic it should also respond that it's in bad health.

### CODE REVIEW

**Comment by bratseth** [comment]
@jonmv @hakonhall fyi

**Comment by jonmv** [comment]
```[INFO] --- abi-check-plugin:7-SNAPSHOT:abicheck (default) @ container-core ---
[ERROR] Class com.yahoo.container.handler.VipStatus: Extra method public void <init>(com.yahoo.container.QrSearchersConfig, com.yahoo.container.handler.ClustersStatus, com.yahoo.container.jdisc.state.StateMonitor)
```

File: container-core/src/main/java/com/yahoo/container/handler/VipStatus.java [hunk 1]
```diff
@@ -24,20 +24,25 @@ public VipStatus() {
...
-    public VipStatus(QrSearchersConfig dispatchers, VipStatusConfig vipStatusConfig, ClustersStatus clustersStatus) {
+    public VipStatus(QrSearchersConfig dispatchers, ClustersStatus clustersStatus) {
         this.clustersStatus = clustersStatus;
-        clustersStatus.setReceiveTrafficByDefault(vipStatusConfig.initiallyInRotation());
         clustersStatus.setContainerHasClusters(! dispatchers.searchcluster().isEmpty());
     }
 
+    /** @deprecated don't pass VipStatusConfig */
+    @Deprecated // TODO: Remove on Vespa 8
+    public VipStatus(QrSearchersConfig dispatchers, VipStatusConfig vipStatusConfig, ClustersStatus clustersStatus) {
```
Comment (gjoranv) [inline container-core/src/main/java/com/yahoo/container/handler/VipStatus.java]: Rename vipStatusConfig arg to 'ignored', so it's easily seen when used in the IDE

File: container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java [hunk 2]
```diff
@@ -15,6 +15,8 @@
  * will survive reconfiguration events and inform other components even immediately after a reconfiguration
  * (where the true statue of clusters may not yet be available).
  *
+ * This is multithread safe.
```
Comment (gjoranv) [inline container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java]: We already use the `javax.annotation.concurrent.ThreadSafe` annotation in 4 modules, which is easier to grep for than (hread safe or threadsafe or multithread safe or all misspellings)

**Review by gjoranv — APPROVED** [review]
Fix build and please consider comments.

File: container-core/src/main/java/com/yahoo/container/handler/VipStatus.java (hunk 1 above)
Comment (bratseth) [reply container-core/src/main/java/com/yahoo/container/handler/VipStatus.java]: Ok, done.

File: container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java (hunk 2 above)
Comment (bratseth) [reply container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java]: I can't imagine why anyone would grep for it

**Pull request merged**